packages/format-currency/src/index.ts

```diff
@@ -1,365 +1,129 @@
-					return;
-				case 'minusSign':
-					sign = '-' as CurrencyObject[ 'sign' ];
-					return;
-				case 'plusSign':
-					sign = '+' as CurrencyObject[ 'sign' ];
-					return;
-			}
-		} );
-
-		const hasNonZeroFraction = ! Number.isInteger( numberAsFloat ) && hasDecimalBeenSet;
-
-		return {
-			sign,
-			symbol,
-			symbolPosition,
-			integer,
-			fraction,
-			hasNonZeroFraction,
-		};
-	}
+interface WithGeoCountry {
+	country_short: string;
+}
 
-	function getValidCurrency( code: string ): string {
-		if ( ! doesCurrencyExist( code ) ) {
+/**
+ * If the user is inside the US using USD, they should only see `$` and not `US$`.
+ */
+async function geolocateCurrencySymbol(): Promise< void > {
+	const geoData = await globalThis
+		.fetch?.( GEO_LOCATION_ENDPOINT_URL )
+		.then( ( response ) => response.json() )
+		.catch( ( error ) => {
+			// Do nothing if the fetch fails.
 			// eslint-disable-next-line no-console
-			console.warn(
-				`getCurrencyObject was called with a non-existent currency "${ code }"; falling back to ${ fallbackCurrency }`
-			);
-			return fallbackCurrency;
-		}
-		return code;
-	}
-
-	function getCurrencyOverride( code: string ): CurrencyOverride | undefined {
-		if ( code === 'USD' && geoLocation !== '' && geoLocation !== 'US' ) {
-			return { symbol: 'US$' };
-		}
-		return defaultCurrencyOverrides[ code ];
-	}
+			console.warn( 'Fetching geolocation for format-currency failed.', error );
+		} );
 
-	function doesCurrencyExist( code: string ): boolean {
-		return Boolean( getCurrencyOverride( code ) );
-	}
+	geoLocation =
+		'string' === typeof ( geoData as WithGeoCountry )?.country_short ? geoData.country_short : '';
+}
 
-	/**
-	 * Set a default locale for use by `formatCurrency` and `getCurrencyObject`.
-	 *
-	 * Note that this is global and will override any browser locale that is set!
-	 * Use it with care.
-	 */
-	function setDefaultLocale( locale: string | undefined ): void {
-		defaultLocale = locale;
+/**
+ * Retrieves the currency override for a given currency.
+ * If the currency is USD and the user is not in the US, it will return `US$`.
+ */
+function getCurrencyOverride( currency: string ): CurrencyOverride | undefined {
+	if ( currency === 'USD' && geoLocation !== '' && geoLocation !== 'US' ) {
+		return { symbol: 'US$' };
 	}
+	return defaultCurrencyOverrides[ currency ];
+}
 
-	function getPrecisionForLocaleAndCurrency(
-		locale: string,
-		currency: string
-	): number | undefined {
-		const formatter = getFormatter( 0, currency, { locale } );
-		return formatter.resolvedOptions().maximumFractionDigits ?? 3; // 3 is the default for Intl.NumberFormat if minimumFractionDigits is not set
+/**
+ * Returns a valid currency code based on a shortlist of currency codes.
+ * Only currencies from the shortlist are allowed. Everything else will fall back to `FALLBACK_CURRENCY`.
+ */
+function getValidCurrency( currency: string ): string {
+	if ( ! getCurrencyOverride( currency ) ) {
```

Comment:
@sirbrillig what were the reasons for having these currency overrides? Meaning, outside of being a list of overrides for particular cases, it's also an actual short-list (can't use anything outside of these).

Reply:
I'm not sure that was by design. Before I refactored this code into its current state, the list of currencies originated because whoever wrote our currency formatting in the first place didn't have access to any libraries (`Int.NumberFormat` didn't yet exist or didn't have currency support) so the formatting decisions were entirely hardcoded based on currency. If there was no entry in the currency list, there would be no way to format it. We could probably safely fall back to `Intl.NumberFormat` now.

Reply:
Got it. I think this was my initial impression too - something that happened before Intl.NumberFormat and stayed that way. I haven't looked at the logs to study the story behind this package. Thanks for confirming. We can return to it at a later time.